src/main/java/org/truffleruby/core/string/ConvertBytes.java

```diff
@@ -288,7 +288,7 @@ private long stringToLong(int nptr, int[] endptr, int base) {
         return 0;
     }
 
-    @TruffleBoundary
+    @TruffleBoundary(transferToInterpreterOnException = false)
```

Comment:
FWIW, I think it is possible to have the cake and eat it too :)
`transferToInterpreterOnException = false` + on the caller profile for the exception like:
```java
final BranchProfile exceptionProfile = BranchProfile.create();

try {
  return byteListToInum();
} catch (RaiseException e) {
  exceptionProfile.enter();
  throw e;
}
```
It might not be worthwhile it here though.